client-docs/sync.md

```diff
@@ -0,0 +1,97 @@
+# Sync API
+
+## Table of Contents
+
+- [Description](#description)
+
+- [Types](#types)
+
+  - [`ConnectionType`](#connectiontype)
+  - [`SyncInfo`](#syncinfo)
+
+- [Methods](#methods)
+
+  - [`info`](#info)
+  - [`setDiscovery`](#setdiscovery)
+  - [`setSync`](#setsync)
+
+## Description
+
+Provides an imperative API for managing project discovery and sync strategies.
+
+## Types
+
+### `ConnectionType`
+
+Represents the connection type used for discovery or sync.
+
+```ts
+type ConnectionType = "lan" | "internet";
+```
+
+### `SyncInfo`
+
+Information about the kind of sync that is enabled.
+
+```ts
+type SyncInfo = {
+  discovery: ConnectionType[];
+  sync: ConnectionType[];
+};
+```
+
+## Methods
+
+### `info`
+
+`() => Promise<SyncInfo>`
+
+Get information about discovery and sync connection types that are enabled.
+
+```ts
+// logs "{ discovery: [], sync: [] }"
+console.log(await client.$sync.info());
+
+await client.$sync.enableDiscovery(["internet", "lan"]);
+
+// logs "{ discovery: ["internet", "lan"], sync: [] }"
+console.log(await client.$sync.info());
+```
+
+### `setDiscovery`
+
+`(connectionTypes: ConnectionType[] | null) => Promise<void>`
+
+Set the discovery connection types to enable. If `setDiscovery` has not been previously called with a valid connection type, discovery becomes enabled. If `connectionTypes` is `null` or `[]`, discovery becomes disabled. Do not rely on the resolving of this method to know when the process starts or stops. Instead, listen to the `'discovery-start'` and `'discovery-stop'` events.
```

Comment:
Is there documentation for `discovery-start` and `discovery-stop` events?

Reply:
yeah all of the events are spec'd in https://github.com/digidem/mapeo-core-docs/pull/9, with context for why they're not documented in each of these scoped docs